dcos/http.py

```diff
@@ -180,3 +229,70 @@ def delete(url, **kwargs):
 def silence_requests_warnings():
     """Silence warnings from requests.packages.urllib3.  See DCOS-1007."""
     requests.packages.urllib3.disable_warnings()
+
+
+def _get_basic_auth_credentials(username):
+    """Get username/password for basic auth
+
+    :param username: username user for authentication
+    :type username: str
+    :returns: HTTPBasicAuth
+    :rtype: requests.auth.HTTPBasicAuth
+    """
+
+    if username is None:
+        sys.stdout.write('Username: ')
+        sys.stdout.flush()
+        username = sys.stdin.readline().strip().lower()
```

Comment:
Should we also print the URL (hostname and realm)? So that the user knows where we are sending those credentials?